### PR TITLE
MBartForConditionalGeneration

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds MBartForConditionalGeneration. Regarding #6416 

@sshleifer

### CODE REVIEW

**Comment by patil-suraj** [comment]
The failure is coming from `test_modeling_tf_electra.py`

**Review by sshleifer — APPROVED** [review]
LGTM, thanks suraj!

File: docs/source/model_doc/mbart.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@sshleifer
+
+Overview
+~~~~~~~~~~~~~~~~~~~~~
+The MBart model was presented in `Multilingual Denoising Pre-training for Neural Machine Translation <https://arxiv.org/abs/2001.08210>`_ by Yinhan Liu, Jiatao Gu, Naman Goyal, Xian Li, Sergey Edunov
+Marjan Ghazvininejad, Mike Lewis, Luke Zettlemoyer. According to the abstract,
+
+MBART is a sequence-to-sequence denoising auto-encoder pre-trained on large-scale monolingual corpora in many languages using the BART objective. mBART is one of the first methods for pre-training a complete sequence-to-sequence model by denoising full texts in multiple languages, while previous approaches have focused only on the encoder, decoder, or reconstructing parts of the text.
+
+The Authors' code can be found `here <https://github.com/pytorch/fairseq/tree/master/examples/mbart>`_
```
Comment (sgugger) [inline docs/source/model_doc/mbart.rst:14]:
```suggestion
The Authors' code can be found `here <https://github.com/pytorch/fairseq/tree/master/examples/mbart>`__
```
(Using only one _ will cause sphinx to associate every here with that link or complain, so it's best to always use two _)

File: src/transformers/modeling_mbart.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from .modeling_bart import BartForConditionalGeneration
+
+
+_CONFIG_FOR_DOC = "MBartConfig"
+_TOKENIZER_FOR_DOC = "MBartTokenizer"
+
+MBART_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "facebook/mbart-large-cc25",
+    "facebook/mbart-large-en-ro",
+    # See all BART models at https://huggingface.co/models?filter=mbart
```
Comment (sgugger) [inline src/transformers/modeling_mbart.py:12]:
```suggestion
    # See all multilingual BART models at https://huggingface.co/models?filter=mbart
```

File: src/transformers/modeling_mbart.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+]
+
+MBART_START_DOCSTRING = r"""
+
+    This model is a PyTorch `torch.nn.Module <https://pytorch.org/docs/stable/nn.html#torch.nn.Module>`_ sub-class.
```
Comment (sgugger) [inline src/transformers/modeling_mbart.py:17]:
```suggestion
    This model is a PyTorch `torch.nn.Module <https://pytorch.org/docs/stable/nn.html#torch.nn.Module>`__ sub-class.
```

**Review by sgugger — APPROVED** [review]
Great work! Just left a few doc nits.

**Comment by patil-suraj** [comment]
Thanks @sgugger , I've applied the suggestions.

File: utils/check_repo.py [hunk 4]
```diff
@@ -30,6 +30,7 @@
     "test_modeling_tf_xlm_roberta.py",
     "test_modeling_xlm_roberta.py",
     "test_modeling_pegasus.py",
+    "test_modeling_mbart.py",
```
Comment (LysandreJik) [inline utils/check_repo.py:33]: Looking forward to adding this to the common tests!

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Pull request merged**